crates/solidity/src/zksolc/main.rs

```diff
@@ -190,18 +194,36 @@ fn main_inner() -> anyhow::Result<()> {
         );
     } else if arguments.output_assembly || arguments.output_binary {
         for (path, contract) in build.contracts.into_iter() {
+            let bytescode = contract.build.bytecode;
+
             if arguments.output_assembly {
-                println!(
-                    "Contract `{}` assembly:\n\n{}",
-                    path, contract.build.assembly_text
-                );
+                let program_blob = match ProgramBlob::parse(bytescode.as_slice()) {
+                    Ok(blob) => blob,
+                    Err(error) => {
+                        bail!("Failed to parse program blob: {}", error);
+                    }
+                };
```

Comment:
Total nitpick. But you can do instead:

```suggestion
                let program_blob = ProgramBlob::parse(bytescode.as_slice())
                    .map_err(|error| anyhow::anyhow!("Failed to parse program blob: {error}"))?;
```

More concise and I personally find it easier to read. But it's a question of style and personal preference so I'll leave it up to you.